packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -552,6 +552,29 @@ abstract class BindingBase {
     });
   }
 
+  // TODO(goderbauer): Remove the next two members after the service/scheduler dependencies
```

Comment:
I presume this will be moved to ServiceBinding?

Reply:
https://github.com/flutter/flutter/pull/54212/commits/43997650bf7abe2d9f891d6a60586e45ba48a0ac shows what this will look like when this is removed.

Reply:
Yes, I was a bit confused why we leave handleAppLifecycleStateChanged and lifecycleState getter in schedulebinding while all other lifecycle related stuffs get moved to servicebinding?

Reply:
Yes. The API lifecycle API stays completely in the scheduling binding to avoid another breaking change. What moves into the service layer is the communication of lifecycle state between the engine and the scheduling layer.

Reply:
Do we have plan to clean this up? I feel it is a bit obscure to structure the code this way.

Reply:
Yes. As soon as this has rolled into google, I will clean up the internal custom bindings and then I'll revert the commit with this change (https://github.com/flutter/flutter/commit/0089896c6803fb079c1713cb2898d9dbfa56b06e). I can't migrate our customers without this intermediate step though.

Reply:
Oh, I meant moving the whole lifecycle related method and api entirely to service binding.

Reply:
I currently don't have plans to change this further to avoid yet another break. The current responsibilities also seem rather clean to me: The SchedulerBinding implements the user facing lifecycle API. It also exposes a method that other bindings can use to inform the scheduler of the current life cycle. The ServiceBinding knows how to retrieve the current lifecycle from the engine and uses that method to inform the scheduler binding about it.

Reply:
sounds good